### DISCUSSION TITLE
Any way to run custom SBT tasks from Bloop?

### DISCUSSION THREAD

**Discussion by alexklibisz** [scalacenter/bloop#1831] [Q&A] [answered]
I know I can import an SBT project to bloop. Now I'm wondering if bloop offers a way to run custom SBT tasks. For example, could I run the `scalafmtAll ` task from the sbt-scalafmt plugin?

**Comment by tgodzik** [comment, marked as answer]
Thanks for raising the discussion! Bloop doesn't allow running any sbt tasks since Bloop is separate from any specific build tool and focuses solely on compilation. We don't plan to change the behaviour any time soon.

**Comment by kpodsiad** [comment]
Workaround just for `scalafmt` case - you could use [coursier bootstrap](https://get-coursier.io/docs/cli-bootstrap#bootstraps) to create "executable" scalafmt jar. That trick is used for instance in [bloop repo](https://github.com/scalacenter/bloop/blob/main/bin/scalafmt)

**Comment by alexklibisz** [comment]
Thanks. I'm also going to look into using the sbt server/client as described here: https://www.youtube.com/watch?v=mWa76dQ7TSo

My main goal is just to minimize the sbt boot times in CI.

**Reply by julienrf** [reply]
Using the sbt server is indeed the recommended way to address your need. Please report any issue you may encounter on the sbt repository and we will help you.
